src/Raven.Server/Documents/Sharding/Handlers/Admin/ShardedAdminShardingHandler.cs

```diff
@@ -13,5 +27,177 @@ public async Task ExecuteMoveDocuments()
+                var (raftIndex, _) = await ServerStore.SendToLeaderAsync(cmd);
+
+                await DatabaseContext.ServerStore.WaitForExecutionOnRelevantNodesAsync(context, shardingConfiguration.Orchestrator.Topology.Members, raftIndex);
+
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NoContent;
+            }
+        }
+
+        [RavenShardedAction("/databases/*/admin/sharding/prefixed/delete", "DELETE")]
+        public async Task DeletePrefixedShardingSetting()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), GetType().Name);
+                var setting = JsonDeserializationCluster.PrefixedShardingSetting(json);
+
+                var shardingConfiguration = ServerStore.Cluster.ReadShardingConfiguration(DatabaseName);
+                bool found = shardingConfiguration.Prefixed.BinarySearch(setting, PrefixedSettingComparer.Instance) >= 0; 
+                if (found == false)
+                    throw new InvalidDataException($"Prefix '{setting.Prefix}' wasn't found in sharding configuration");
+
+                var clusterTopology = ServerStore.GetClusterTopology(context);
+                var urls = shardingConfiguration.Orchestrator.Topology.Members.Select(clusterTopology.GetUrlFromTag).ToArray();
+
+                if (await AssertNoDocumentsStartingWith(context, setting.Prefix, urls) == false)
+                    throw new InvalidOperationException(
+                        $"Cannot remove prefix '{setting.Prefix}' from {nameof(ShardingConfiguration)}.{nameof(ShardingConfiguration.Prefixed)}. " +
+                        $"There are existing documents in database '{DatabaseName}' that start with '{setting.Prefix}'. " +
+                        "In order to remove a sharding by prefix setting, you cannot have any documents in the database that starts with this prefix.");
+
+                var cmd = new DeletePrefixedShardingSettingCommand(setting, DatabaseName, GetRaftRequestIdFromQuery());
+                var (raftIndex, _) = await ServerStore.SendToLeaderAsync(cmd);
+
+                await DatabaseContext.ServerStore.WaitForExecutionOnRelevantNodesAsync(context, shardingConfiguration.Orchestrator.Topology.Members, raftIndex);
+
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NoContent;
+            }
+        }
+
+        [RavenShardedAction("/databases/*/admin/sharding/prefixed/update", "POST")]
+        public async Task UpdatePrefixedShardingSetting()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), GetType().Name);
```

Comment:
Here too, for that matter. Can move that higher before the transaction starts

Reply:
fixed